### ISSUE TITLE
增量训练中添加optimizer后报错

### ISSUE THREAD

**Issue by leaf918** [PaddlePaddle/models#2903]
` with fluid.program_guard(self.train_program, self.startup_program):
            self.image = fluid.layers.data(name='image', shape=[3, 224, 224], dtype='float32')
            self.label = fluid.layers.data(name='label', shape=[1], dtype='int64')
            # model definition ++++++++++++++++++++++++++++++++++++++++++
            model = models.__dict__[model_name]()
            out = model.net(input=self.image, class_dim=class_dim)  # 概率数组
            #
            cost, pred = fluid.layers.softmax_with_cross_entropy(
                out,
                self.label,
                return_softmax=True
            )

            avg_cost = fluid.layers.mean(x=cost)
            acc_top1 = fluid.layers.accuracy(input=pred,
                                             label=self.label,
                                             k=1)
            acc_top5 = fluid.layers.accuracy(input=pred,
                                             label=self.label,
                                             k=5)
            optimizer = fluid.optimizer.SGD(learning_rate=0.01)
            optimizer.minimize(avg_cost)
            self.parallel_program = compiler.CompiledProgram(self.train_program)
            self.parallel_program.with_data_parallel(loss_name=avg_cost.name)
            # load pretrained vars,load data to default main program
            self.exe = fluid.Executor(self.exe_place)
            fluid.io.load_persistables(
                self.exe,
                output_dir if self.mode == 'inference' else pretrain_dir,
                main_program=self.startup_program
            )
`

``
       
![image](https://user-images.githubusercontent.com/170311/61711063-c4788880-ad85-11e9-85b6-dadf662bbce8.png)

**Comment by leaf918** [comment]
![image](https://user-images.githubusercontent.com/170311/61713049-44085680-ad8a-11e9-8188-f29a4a847578.png)
加入红框的两句后报错，应该怎么处理？
![image](https://user-images.githubusercontent.com/170311/61713073-4ec2eb80-ad8a-11e9-8c54-853609dba044.png)

**Comment by guoshengCS** [comment]
这个应该是因为加入optimizer后创建了一个`learning_rate_0`的persistable的variable以能够从checkpoint恢复训练，此时使用`load_persistable`恢复训练时要提供`learning_rate_0`，不然无法完全一致的恢复训练，而提供的用于load的文件不包含这个variable，所以报错。
对于增量训练时一些不需要或者因没保存而无法恢复内容，可以使用[`load_vars`](https://www.paddlepaddle.org.cn/documentation/docs/zh/1.5/api_cn/io_cn.html#load-vars)这个接口，使用时为vars传入需要恢复的内容。

**Comment by leaf918** [comment]
![image](https://user-images.githubusercontent.com/170311/61768337-c3913680-ae19-11e9-9f62-a7ee6d91f29a.png)

如何知道非红框内的变量的列表， @guoshengCS

**Comment by guoshengCS** [comment]
`load_vars`包含一个predict的参数，可以传入一个function，另这个function判断`var.name=='learning_rate_0'`

**Comment by leaf918** [comment]
![image](https://user-images.githubusercontent.com/170311/61775549-b715d980-ae2b-11e9-9384-95f6887b4bc0.png)
not working...

**Comment by guoshengCS** [comment]
如果load_vars提供了vars这个参数，predict参数就会被忽略了，不用同时设置，另外这个报错和之前不一样了吧，之前是load有问题
